pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/table/DynamicPulsarSerializationSchema.java

```diff
@@ -125,6 +133,9 @@ public void serialize(RowData consumedRow, TypedMessageBuilder<RowData> messageB
             return;
         }
 
+        // set delay message.
+        messageBuilder.deliverAfter(delayMilliseconds, TimeUnit.MILLISECONDS);
```

Comment:
It should be judged that `delayMilliseconds` is greater than 0 before it can be set.